modules/atomic-widgets/validators/styles-validator.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Elementor\Modules\AtomicWidgets\Validators;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Styles_Validator {
+	const VALID_STATES = [
+		'hover',
+		'active',
+		'focus',
+		null,
+	];
+
+	private array $schema;
+
+	private array $errors_bag = [];
+
+	public function __construct( array $schema ) {
+		$this->schema = $schema;
+	}
+
+	public static function make( array $schema ): self {
+		return new static( $schema );
+	}
+
+	/**
+	 * @param array $styles
+	 * The key of each item represents the style id,
+	 * and the value is the style object to validate
+	 *
+	 * @return array{
+	 *     0: bool,
+	 *     1: array<string, mixed>,
+	 *     2: array<string>
+	 * }
+	 */
+	public function validate( array $styles ): array {
+		foreach ( $styles as $style_id => $style ) {
+			if ( ! isset( $style['id'] ) || ! is_string( $style['id'] ) ) {
+				$this->errors_bag[] = 'id';
+				continue;
+			}
```

Comment:
If it failed but I still have access to the styles in the return value, why did we validated from the first place? lets talk about it